central/signatureintegration/datastore/datastore_impl.go

```diff
@@ -149,6 +164,93 @@ func (d *datastoreImpl) verifyIntegrationIDDoesNotExist(ctx context.Context, id
 	return nil
 }
 
+func (d *datastoreImpl) verifyIntegrationIDIsNotInPolicy(ctx context.Context, id string) error {
+	policyCtx := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Policy)))
+
+	policies, err := d.policyStore.GetAllPolicies(policyCtx)
+	if err != nil {
+		return errors.Wrap(err, "retrieving all policies")
+	}
+
+	var policiesContainingID []string
+	for _, p := range policies {
+		if checkIfPolicyContainsID(id, p) {
+			policiesContainingID = append(policiesContainingID, p.GetName())
+		}
+	}
+
+	if len(policiesContainingID) != 0 {
+		integration, err := d.getSignatureIntegrationByID(ctx, id)
+		if err != nil {
+			return err
+		}
+
+		// Fetch policies with the user context, we can safely ignore the error since one would be returned in case
+		// the user does not have access to policies.
+		policiesVisibleToUser, _ := d.policyStore.GetAllPolicies(ctx)
```

Comment:
what if error returned is not `sac.ErrResourceAccessDenied`?

Reply:
Handled this now, return any non-nil error thats not `sac.ErrResourceAccessDenied`.